io/src/lib.rs

```diff
@@ -65,6 +80,27 @@ pub struct Take<'a, R: Read + ?Sized> {
     remaining: u64,
 }
 
+impl<'a, R: Read + ?Sized> Take<'a, R> {
+    #[cfg(any(feature = "alloc", feature = "std"))]
+    #[inline]
+    fn read_to_end(&mut self, buf: &mut Vec<u8>) -> Result<usize> {
+        let mut read: usize = 0;
+        let mut chunk = [0u8; 64];
```

Comment:
I would guess something like 1024 would be a better number. I kinda want this to be `BufRead` and just append the contents of `fill_buf`. Would make things sooo much better... Sadly specialization is not a thing yet. (Though we could specialize on things like slices and cursors.)

Reply:
Let's fight about this constant in a followup PR :). Intuitively 1024 sounds better but if we almost always use this on readers that are buffered anyway I'm not sure it matters (and possibly using a big buffer could cause some sort of bufferbloat).

Reply:
That's exactly why specialization would be great. But this method is public, on `Read` so users could call it on non-buffered stuff (treating the crate like it's `core3`).

And for use in PSBT, which is the only place where we do use it, `push_decode` again. :)